llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp

```diff
@@ -1492,23 +1493,42 @@ bool HWAddressSanitizer::instrumentStack(memtag::StackInfo &SInfo,
   return true;
 }
 
+static void emitRemark(const Function &F, OptimizationRemarkEmitter &ORE,
+                       bool Skip) {
+  if (Skip) {
+    ORE.emit([&]() {
+      return OptimizationRemark(DEBUG_TYPE, "Skip", &F)
+             << "Skipped: F=" << ore::NV("Function", &F);
+    });
+  } else {
+    ORE.emit([&]() {
+      return OptimizationRemarkMissed(DEBUG_TYPE, "Sanitize", &F)
+             << "Sanitized: F=" << ore::NV("Function", &F);
```

Comment:
Why do we need this arg? We already have the F in the output either way.

Now the output looks like this

```
--- !Missed
Pass:            hwasan
Name:            Sanitize
Function:        test_retptr
Args:
  - String:          'Sanitized: F='
  - Function:        test_retptr
...

I suggest just making this `OptimizationRemarkMissed(DEBUG_TYPE, "Sanitize", &F);` Same above
```

Reply:
Also, wouldn't it be better to have the same name for the Skip and non skip-Cases, and just distinguish by the Miss / Pass